numba/cuda/codegen.py

```diff
@@ -113,6 +115,17 @@ def _get_ptxes(self, cc=None):
         arch = nvvm.get_arch_option(*cc)
         options = self._nvvm_options.copy()
         options['arch'] = arch
+        if not nvvm.NVVM().is_nvvm70:
+            # Avoid enabling debug for NVVM 3.4 as it has various issues. We
+            # need to warn the user that we're doing this if any of the
+            # functions that they're compiling have `debug=True` set, which we
+            # can determine by checking the NVVM options.
+            debug_callee = any(lib._nvvm_options.get('debug')
+                               for lib in self.linking_libraries)
+            if options.get('debug') or debug_callee:
+                msg = "debuginfo is not generated for CUDA versions < 11.2"
```

Comment:
From OOB discussion, this message is going to be updated to include the function name so it's easy for users to find the function(s) causing this warning.